### PR TITLE
[14.0][ADD] partner_category_security: New addon.

### PR DESCRIPTION
**Pull request by victoralmau**
New addon.

Please @pedrobaeza and @CarlosRoca13 can you review it?

@Tecnativa TT38576

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]
Code review :+1:

File: partner_category_security/README.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_category_security
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/134/14.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module adds permissions to manage contact labels.
```
Comment (pedrobaeza) [inline partner_category_security/README.rst]:
```suggestion
This module adds a specific security group to manage contact tags (categories).
```

File: partner_category_security/README.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+#. *Contact Tags > Publisher* permission allows you to modify contact tags.
```
Comment (pedrobaeza) [inline partner_category_security/README.rst]:
```suggestion
#. *Contact Tags > Publisher* permission allows you to modify the tags assigned to contacts.
```

File: partner_category_security/README.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+#. *Contact Tags > Manager* permission also allows you to create and delete contact tag records.
```
Comment (pedrobaeza) [inline partner_category_security/README.rst]:
```suggestion
#. *Contact Tags > Manager* permission also allows you to manage such tag (create, modify or remove them).
```

File: partner_category_security/README.rst [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+#. *Contacts > Configuration > Contact Tags* menu will only be visible if you have one of these permissions.
```
Comment (pedrobaeza) [inline partner_category_security/README.rst]: It should be only available if the latest one is granted.

File: partner_category_security/README.rst [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+#. *Extra Rights / Contact Creation* permission now only allows reading contact tag records.
```
Comment (pedrobaeza) [inline partner_category_security/README.rst]: I think all internal users should have read permission.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by victoralmau** [comment]
Changes done

**Comment by pedrobaeza** [comment]
I have tried in runboat, and I'm able to change tags in the partner with no permission assigned.

**Comment by victoralmau** [comment]
> I have tried in runboat, and I'm able to change tags in the partner with no permission assigned.

Ok, I guess you are testing with the demo user. 

You will need to add a glue module with `crm` (`partner_category_security_crm`) because this rule https://github.com/odoo/odoo/blob/14.0/addons/crm/security/ir.model.access.csv#L7 gives permissions to do so and you will need to overwrite it.

Do you think I should add that new addon in this PR?

**Comment by pedrobaeza** [comment]
I don't think it's something of the CRM glue module. It's simply that the field is not readonly for that group. You have to modify the view and add 2 times the field: one only readonly and available for the basic internal user, and other not readonly if you have the other permission.

**Comment by victoralmau** [comment]
> I don't think it's something of the CRM glue module. It's simply that the field is not readonly for that group. You have to modify the view and add 2 times the field: one only readonly and available for the basic internal user, and other not readonly if you have the other permission.

Ok, you're right, I hadn't thought of that approach. It is already added.

File: partner_category_security/views/res_partner_view.xml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+                >partner_category_security.group_partner_category_user</attribute>
+            </field>
+            <!-- Add the field again (read-only) to make it visible to other users !-->
+            <field name="category_id" position="after">
+                <field
+                    name="category_id"
+                    widget="many2many_tags"
+                    options="{'color_field': 'color'}"
+                    placeholder="Tags..."
+                    readonly="1"
```
Comment (pedrobaeza) [inline partner_category_security/views/res_partner_view.xml:20]: Don't break any automatism on onchanges.

```suggestion
                    readonly="1"
                    force_save="1"
```

**Comment by victoralmau** [comment]
Changes done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
/ocabot merge nobump

Please proceed with the CRM extension

**Pull request merged**